File: frontend/src/interfaces/article.ts
export type Article = {
  title: string;
  shortDescription: string;
  fullContent: string;
  author: string;
  publishedDate: string;
  tags: string[];
};

export type GetArticle = Article & { articleID: string };

export type FieldValues = {
  title: string;
  shortDescription: string;
  fullContent: string;
  author: string;
  publishedDate: string;
  tags: string;
};

File: ArticlesApiPostgres/Models/Article.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ArticlesApiPostgres.Models
{
    public class Article
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int ArticleID { get; set; }
        public string Title { get; set; }
        public string ShortDescription { get; set; }
        public string FullContent { get; set; }
        public string Author { get; set; }
        public DateTime PublishedDate { get; set; }
        public ICollection<ArticleTag> ArticleTags { get; set; } = new List<ArticleTag>();
    }
}